---
title: "Structural Bioinformatics"
author: "A59026686"
format: pdf
---

#1. Introduction to RCSB Protein Data Bank (PDB)
```{r}
data <- read.csv("Data Export Summary.csv", row.names = 1)
data
```

At the time of writing, there are 183,201 protein structures.
In Uniport, there are 251600,768 protein sequences.
```{r}
round(183201/251600768*100,2)
```
Q1: What percentage of structures in the PDB are solved by X-Ray and Electron Microscopy.
```{r}
total <- data["Protein (only)", "Total"]
total <- as.numeric(gsub(",", "", total))

total_xray_em <- as.numeric(gsub(",","",data["Protein (only)", "X.ray"])) +
                      as.numeric(gsub(",","",data["Protein (only)", "EM"]))           

perc <- round(total_xray_em/total*100,2)
perc
```

Q2: What proportion of structures in the PDB are protein?

```{r}
all_structures <- as.numeric(gsub(",","",data[,"Total"]))
all_structures

prop_proteins <- round(all_structures[1]/sum(all_structures)*100,2)
prop_proteins
```
Making a function to remove commas
```{r}
rm.comma <- function(x){
  as.numeric(gsub(",","",x))
}

pdbstats <- apply(data, 2, rm.comma)
```

Will add the rownames from the origianl table.
```{r}
rownames(pdbstats) <- rownames(data)
pdbstats
```


Q3: Type HIV in the PDB website search box on the home page and determine how many HIV-1 protease structures are in the current PDB?

There are 200 HIV-1 protease structures.

Here is a pic of HIV-Pr.

![](1HSG.png)
And a nicer pic is:


![A lovely Image](1HSG_2.png)




Q4: Water molecules normally have 3 atoms. Why do we see just one atom per water molecule in this structure?

The resolution at which the structure of 1HSG was resolved is 2A but hydrgen atoms are of the size 0.5A. Hence, water molecules appear as just 1 atom.

Q5: There is a critical “conserved” water molecule in the binding site. Can you identify this water molecule? What residue number does this water molecule have

HOH 308

#3. Introduction to Bio3D

```{r}
library(bio3d)
```

Reading PDB data file into R:
```{r}
pdb <- read.pdb("1hsg")
pdb
```
Q7: How many amino acid residues are there in this pdb object? 
198

Q8: Name one of the two non-protein residues?
HOH, MK1

Q9: How many protein chains are in this structure?
2

Observing the attributes:
```{r}
attributes(pdb)
```
```{r}
head(pdb$atom)
```
```{r}
head(pdb$atom$resid)
```
```{r}
aa321(pdb$atom$resid[pdb$calpha])
```




Predicting functional motions of a single structure:
```{r}
adk <- read.pdb("6s36")
adk
```

Normal Mode Analysis:
```{r}
m <- nma(adk)
plot(m)
```
View the movie:
```{r}
mktrj(m, pdb = adk, file="adk_m7.pdb")
```
